CMakeLists.txt

```diff
@@ -795,17 +795,24 @@ set (Seastar_PRIVATE_CXX_FLAGS
   -UNDEBUG
   -Wall
```

Comment:
> In https://github.com/scylladb/seastar/pull/1867, we replaced the `-Wdeprecated-declarations`

Please use commit hash references, not pull requests, to make the commit log less tied to github.